src/EFCore.Relational/Query/RelationalQueryableMethodTranslatingExpressionVisitor.cs

```diff
@@ -1668,7 +1669,181 @@ protected override Expression VisitMethodCall(MethodCallExpression methodCallExp
                     ?? methodCallExpression.Update(null!, new[] { source, methodCallExpression.Arguments[1] });
             }
 
+            if (methodCallExpression.Method.IsGenericMethod
+                && (methodCallExpression.Method.GetGenericMethodDefinition() == QueryableMethods.ElementAt
+                    || methodCallExpression.Method.GetGenericMethodDefinition() == QueryableMethods.ElementAtOrDefault))
+            {
+                source = methodCallExpression.Arguments[0];
+                var selectMethodCallExpression = default(MethodCallExpression);
+
+                if (source is MethodCallExpression sourceMethodCall
+                    && sourceMethodCall.Method.IsGenericMethod
```

Comment:
Patterns FTW (also below):

```suggestion
                if (source is MethodCallExpression { Method.IsGenericMethod: true } sourceMethodCall
```